### PR TITLE
Loaded the data in database

### PR DESCRIPTION
**Pull request by Pratikshadange2005**

### CODE REVIEW

File: load/load.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# Step 1: Extract the zip file into a temporary directory
+def extract_zip_file(zip_file_path):
+    temp_dir = tempfile.TemporaryDirectory()  # Create a temporary directory
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
+        zip_ref.extractall(temp_dir.name)
+    return temp_dir
+
+# Step 2: Load JSON files into DuckDB with 'loaded_at' and process only new files
+def load_json_gz_files_in_duck_db(json_dir, duckdb_path):
+    # Open the DuckDB connection (ensure you connect to the correct persistent database)
```
Comment (edsioufi) [inline load/load.py]: Style: I'd suggest avoiding comments unless where necessary. In this case, the code below is self-explanatory. Adding all these comments can make the code less readable than just letting the code speak for itself. This apply to other comments as well that we could remove!

File: load/load.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import os
+import zipfile
+import duckdb
+import tempfile
+from datetime import datetime
+
+# Step 1: Extract the zip file into a temporary directory
+def extract_zip_file(zip_file_path):
```
Comment (edsioufi) [inline load/load.py]: Currently this function returns a temp_dir without cleaning it up, which leads to confusing behavior in the main function. To use this function properly, you need to know that this function has a "side-effect" (meaning an unwanted side-effect behavior) of creating a temp folder without destroying it.

A better way woudl be to pass the target directory as second argument to this function, and let the main function caller handle creating and cleaning up the temp_dir there in the same context.

File: load/load.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+# Main function 
+if __name__ == "__main__":
+    # path to the zip file
+    zip_file_path = os.path.join(os.path.dirname(__file__), "..", "data", "gharchive_sample.zip")
+    
+    # Define the DuckDB file path (persistent file)
+    duckdb_path = 'github_stars.db'  
+
+    # Step 3: Extract the zip file into a temporary folder
+    temp_dir = extract_zip_file(zip_file_path)
```
Comment (edsioufi) [inline load/load.py]: Instead, create the temp_dir here and pass it to extract_zip_file.

As a bonus, this allows us to [use the "with" syntax](https://stackoverflow.com/a/55104228) which creates and cleans up the temp dir in one go, making it cleaner and more "pythonic".

**Review by edsioufi — CHANGES_REQUESTED** [review]
Awesome work! 🙌 

I dropped a couple of comments that should be quick to address, and take the code to the next level.

File: load/load.py (hunk 1 above)
Comment (Pratikshadange2005) [reply load/load.py]: I also thought this beforehand but just leave them for more clarity.

File: load/load.py (hunk 2 above)
Comment (Pratikshadange2005) [reply load/load.py]: done

File: load/load.py (hunk 3 above)
Comment (Pratikshadange2005) [reply load/load.py]: done

**Review by edsioufi — APPROVED** [review]
Awesome work! Incremental loading looks great 🚀

**Pull request merged**